openedx/core/djangoapps/content_tagging/rest_api/v1/views.py

```diff
@@ -130,8 +145,111 @@ def orgs(self, request, **_kwargs) -> Response:
 class ObjectTagOrgView(ObjectTagView):
     """
     View to create and retrieve ObjectTags for a provided Object ID (object_id).
-    This view extends the ObjectTagView to add Organization filters for the results.
 
     Refer to ObjectTagView docstring for usage details.
     """
     filter_backends = [ObjectTagTaxonomyOrgFilterBackend]
+
+
+class ObjectTagExportView(APIView):
+    """"
+    View to export a CSV with all children and tags for a given object_id.
+    """
+    def get(self, request: Request, **kwargs) -> StreamingHttpResponse:
+        """
+        Export a CSV with all children and tags for a given object_id.
+        """
+
+        def _content_generator(
+            tagged_content: TaggedContent, level: int = 0
+        ) -> Iterator[tuple[TaggedContent, int]]:
+            """
+            Generator that yields the tagged content and the level of the block
+            """
+            yield tagged_content, level
+            if tagged_content.children:
+                for child in tagged_content.children:
+                    yield from _content_generator(child, level + 1)
```

Comment:
I would rename `_content_generator` to `_iterate_with_level` and put it alongside `_build_object_tree_with_objecttags` and the `TaggedContent` type into `rest_api/v1/tag_export_helpers.py`